src/components/SignupForm/interfaces/SignupFormInterfaces.tsx

```diff
@@ -0,0 +1,27 @@
+export interface SignUpFormDataInterface {
+  name: string;
+  email: string;
+  password: string;
+  repeatPassword: string;
+}
+
+export interface FormFieldConfigInterface {
+  label: string;
+  type: string;
+  name: string;
+  isError: boolean;
+  errorText: string;
+  value: string;
+}
+
+export interface ErrorInterface {
+  isError: boolean;
+  errorText: string;
+}
+
+export interface ErrorMessageVisibleInterface {
```

Comment:
SignUpFormDataInterface is tightly connected to ErrorMessageVisibleInterface. Whenever you add or remove from SignUpFormDataInterface something then you should do this here too. That's why mapped types (https://www.typescriptlang.org/docs/handbook/2/mapped-types.html) are really good in this situation. Instead of ErrorMessageVisibleInterface you could do something like this:


You don't need this interface. Just use

  [key in keyof  {
  name: boolean;
  email: boolean;
  password: boolean;
  repeatPassword: boolean;
}]: string;
};




  [Property in keyof  Credentials & {repeatPassword: string}]: boolean